pkg/apis/core/helper/helpers.go

```diff
@@ -167,3 +169,61 @@ func FindWorkerByName(workers []core.Worker, name string) *core.Worker {
 	}
 	return nil
 }
+
+// GetRemovedVersions finds versions that have been removed in the old compared to the new version slice.
+// returns a map associating the index in the old version slice with the version and a string set containing the versions.
+func GetRemovedVersions(old []core.ExpirableVersion, new []core.ExpirableVersion) (map[int]core.ExpirableVersion, sets.String) {
```

Comment:
I think we should avoid using multiple return values for returning redundant data. It's kind of hard to understand why these different "collections" are returned and only because you need exactly a `map` here whereas you need a `set` there shouldn't be the reason.

Reply:
adjusted it.